ts/parts-more/Polar.ts

```diff
@@ -607,82 +637,341 @@ if (seriesTypes.column) {
+                                end = yAxis.translate(stackValues[1]);
+
+                                // If starting point is beyond the
+                                // range, set it to 0
+                                if (defined(start)) {
+                                    start = U.clamp(start, 0, visibleRange);
+                                }
+                            }
+                        }
+                    } else {
+                        // Initial start and end angles for radial bar
+                        start = thresholdAngleRad;
+                        end = point.plotY;
+                    }
+
+                    if (start > end) {
+                        // Swapping start and end
+                        end = [start, start = end][0];
+                    }
+
+                    // Prevent from rendering point outside the
+                    // acceptable circular range
+                    if (!reversed) {
+                        if (start < yMin) {
+                            start = yMin;
+                        } else if (end > yMax) {
+                            end = yMax;
+                        } else if (end < yMin || start > yMax) {
+                            start = end = 0;
+                        }
+                    } else {
+                        if (end > yMin) {
+                            end = yMin;
+                        } else if (start < yMax) {
+                            start = yMax;
+                        } else if (start > yMin || end < yMax) {
+                            start = end = visibleRange;
+                        }
+                    }
+
+                    if ((yAxis.min as any) > (yAxis.max as any)) {
+                        start = end = reversed ? visibleRange : 0;
+                    }
+
+                    start += startAngleRad;
+                    end += startAngleRad;
+
+                    // In case when radius, inner radius or both are
+                    // negative, a point is rendered but partially or as
+                    // a center point
+                    innerR = Math.max(barX, 0);
+                    r = Math.max(barX + point.pointWidth, 0);
+
+                    // Required for the pie animation
+                    point.startR = start === end ? 0 : r;
+                    point.shapeArgs = shapeArgs = {
+                        x: center[0],
+                        y: center[1],
+                        r: r,
+                        innerR: innerR,
+                        start: start,
+                        end: end
+                    };
+
+                    // A correct value for stacked or not fully visible
+                    // point
+                    point.plotY = (defined(series.translatedThreshold) &&
+                        (start < series.translatedThreshold ? start : end)) -
+                            startAngleRad;
+
+                    if (point.graphic &&
```

Comment:
This part should be refactored out of the `translate` function. In general, the translate function should only create the shapeArgs and do processing of positions. All actual rendering and creating/updating of `graphic` should happen in the `drawPoints` function. If necessary, you can create a `drawPoints` event that is fired at the top of the ColumnSeries `drawPoints` function.